src/main/java/seedu/address/logic/commands/EditCommand.java

```diff
@@ -109,6 +136,26 @@ private static Person createEditedPerson(Person personToEdit, EditPersonDescript
                 updatedGroupTags, updatedModuleTags);
     }
 
+    /**
+     * Creates and returns a {@code User} with the details of {@code personToEdit}
+     * edited with {@code editPersonDescriptor}.
+     */
+    private static User createEditedUser(User userToEdit, EditPersonDescriptor editPersonDescriptor) {
+        Name updatedName = editPersonDescriptor.getName().orElse(userToEdit.getName());
+        Phone updatedPhone = editPersonDescriptor.getPhone().orElse(userToEdit.getPhone());
+        Email updatedEmail = editPersonDescriptor.getEmail().orElse(userToEdit.getEmail());
+        Address updatedAddress = editPersonDescriptor.getAddress().orElse(userToEdit.getAddress());
+        TelegramHandle updatedTelegramHandle = editPersonDescriptor.getTelegramHandle()
+                .orElse(userToEdit.getTelegramHandle());
+        Set<GroupTag> updatedGroupTags = editPersonDescriptor.getGroupTags()
+                .orElse(userToEdit.getGroupTags());
+        Set<ModuleTag> updatedModuleTags = editPersonDescriptor.getModuleTags()
+                .orElse(userToEdit.getModuleTags());
+
+        return User.getSingletonUser(updatedName, updatedPhone, updatedEmail,
```

Comment:
We need to consider changing user singleton to allow editing of user information later on.